### PR TITLE
Additional check to skip redeprovisioning

### PR DESCRIPTION
**Pull request by jaroslaw-pieszka**
**Description**

If there is not ongoing and not temporary deprovisioning operation but there are no steps to retry/redo, we skip creation of a new operation

Changes proposed in this pull request:

- refactored check for ongoing, not temporary operation 
- added check for not ongoing, not temporary but with empty set on not completed steps
- added doc for deprovision retriever

**Related issue(s)**

### CODE REVIEW

**Review by wozniakjan — COMMENTED** [review]
this is a refinement of https://github.com/kyma-project/control-plane/pull/2491, correct?

Maybe some test flake. Or some tests need adjustments or there is some subtle difference between how the condition got inverted and how De Morgan would have inverted it

```
[0173] | time="2023-02-20T07:47:12Z" level=warning msg="while executing request: got unexpected status code while calling Kyma Environment Broker: want: 202, got: 500" component="broker/client.go:107:broker.(*Client).Deprovision.func1"
```

File: components/kyma-environment-broker/internal/broker/instance_deprovision.go [hunk 1]
```diff
@@ -95,9 +107,9 @@ func (b *DeprovisionEndpoint) Deprovision(ctx context.Context, instanceID string
 	}, nil
 }
 
-func isOngoingNotTemporaryDeprovisioning(existingOperation *internal.DeprovisioningOperation) bool {
-	return !(existingOperation == nil ||
-		existingOperation.Temporary ||
-		existingOperation.State == domain.Failed ||
-		existingOperation.State == domain.Succeeded)
+func isSucceededNotTemporaryButWithNoSteps(existingOperation *internal.DeprovisioningOperation) bool {
```
Comment (wozniakjan) [inline components/kyma-environment-broker/internal/broker/instance_deprovision.go:110]: nit: the name is quite verbose, how would you feel about something shorter, maybe something along the lines of -
```suggestion
func retriggerable(existingOperation *internal.DeprovisioningOperation) bool {
```

File: docs/kyma-environment-broker/03-16-deprovision-retrigger-cronjob.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+## Details
+
+During regular deprovisioning some steps could be omitted due to the occurrence of some errors. These errors not necessarily cause deprovisioning process to fail.
+Sometimes we ignore not severe, supposedly temporary errors, proceed with deprovisioning and declare process as successful. But there are some not completed steps
+then can be retried later on. We store the list of not completed steps, and mark the deprovisioning operation setting `deletedAt` to current timestamp.
+The Job iterates over the instances and for each with `deletedAt` appropriately set sends a DELETE to Kyma Environment Broker (KEB).  
+
+### Dry-run mode
+
+If you need to test the Job, you can run it in the `dry-run` mode.
```
Comment (IwonaLanger) [inline docs/kyma-environment-broker/03-16-deprovision-retrigger-cronjob.md:14]:
```suggestion
If you need to test the Job, you can run it in the dry-run mode.
```

File: docs/kyma-environment-broker/03-16-deprovision-retrigger-cronjob.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+In that mode, the Job only logs the information about the candidate instances (i.e. instances meeting the configured criteria). The instances are not affected.
```
Comment (IwonaLanger) [inline docs/kyma-environment-broker/03-16-deprovision-retrigger-cronjob.md:15]:
```suggestion
In this mode, the Job only logs the information about the candidate instances (for example, instances meeting the configured criteria). The instances are not affected.
```

File: docs/kyma-environment-broker/03-16-deprovision-retrigger-cronjob.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+By default, the CronJob is set to run every day at 3:00 am:
+```yaml  
+kyma-environment-broker.trialCleanup.schedule: "0 3 * * *"
+```
+
+Use the following environment variables to configure the Job:
+
+| Environment variable | Description                                                                                                               | Default value                            |
+|---|---------------------------------------------------------------------------------------------------------------------------|------------------------------------------|
+| **APP_DRY_RUN** | Specifies whether to run the Job in the [`dry-run` mode](#details).                                                       | `true`                                   |
```
Comment (IwonaLanger) [inline docs/kyma-environment-broker/03-16-deprovision-retrigger-cronjob.md:35]:
```suggestion
| **APP_DRY_RUN** | Specifies whether to run the Job in the [dry-run mode](#details).                                                       | `true`                                   |
```

**Review by IwonaLanger — APPROVED** [review]

File: components/kyma-environment-broker/internal/broker/instance_deprovision.go (hunk 1 above)
Comment (jaroslaw-pieszka) [reply components/kyma-environment-broker/internal/broker/instance_deprovision.go]: Removed

**Review by wozniakjan — APPROVED** [review]

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**